src/wasm/wasm-type.cpp

```diff
@@ -305,9 +253,16 @@ std::unordered_map<TypeInfo, uintptr_t> indices = {
   {TypeInfo({Type::f64}), Type::f64},
   {TypeInfo({Type::v128}), Type::v128},
   {TypeInfo({Type::funcref}), Type::funcref},
+  {TypeInfo(HeapType(HeapType::FuncKind), true), Type::funcref},
   {TypeInfo({Type::externref}), Type::externref},
-  {TypeInfo({Type::nullref}), Type::nullref},
+  {TypeInfo(HeapType(HeapType::ExternKind), true), Type::externref},
+  // TODO (GC): Add canonical ids
+  // * `(ref null any) == anyref`
+  // * `(ref null eq) == eqref`
+  // * `(ref i31) == i31ref`
```

Comment:
Is there anything blocking this? If not, I think you should go ahead with these in this PR.

Reply:
If I add these, I'll have to touch all the code locations switching over basic types, then handling `Type::anyref`, `Type::eqref` and `Type::i31ref` as well. Not opposed to doing that, but it might make this PR's diff somewhat unwieldy? lmk :)

Reply:
Oh right, I forgot about all those switches that would need to be updated. Seems fine to leave these to a follow-on PR, then. Or to just use a macro everywhere to mark them as TODO.

Reply:
Did a quick search for `::nullref` to get an idea. 83 locations, not too bad, but it also looks like there's quite a bit of low hanging fruit that might make sense to implement right away, and I guess that'd be better suited in a separate PR? :)

Reply:
Sounds good to me!